src/redux/rockets/reducer.js

```diff
@@ -14,6 +15,12 @@ const rocketsReducer = (state = [], action) => {
         return { ...rocket, reserved: !rocket.reserved };
       });
 
+    case CANCEL_ROCKET:
+      return state.map((rocket) => {
+        if (rocket.id !== action.payload) return rocket;
+        return { ...rocket, reserved: false };
+      });
+
```

Comment:
Wow, it works very well, but don't you think you increased the amount of code by adding the CANCEL_ROCKET action when it can already change the reserved status on line 15 by !rocket.reserved